cql3/statements/raw/select_statement.hh

```diff
@@ -52,7 +52,7 @@ public:
     class parameters final {
     public:
         using orderings_type = std::vector<std::pair<shared_ptr<column_identifier::raw>, ordering>>;
-        enum class statement_subtype { REGULAR, JSON };
+        enum class statement_subtype { REGULAR, JSON, PRUNE_MATERIALIZED_VIEW };
```

Comment:
This also suggests refactoring is needed: we store the type both in the select_statement real type (the subclass) and here. But we can do that later too, no rush.

Reply:
This patch gives me `CREATE TABLE tab AS SELECT` and `DELETE  FROM tab WHERE ...` vibes.